dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -2193,6 +2237,349 @@ const char* DynamicData::typekind_to_string(TypeKind tk) const
   }
 }
 
+bool print_dynamic_data(OpenDDS::XTypes::DynamicData dd, OpenDDS::XTypes::DynamicType_rch dt, OpenDDS::DCPS::String& type_string, OpenDDS::DCPS::String indent)
+{
+  OpenDDS::DCPS::String member_name;
+  OpenDDS::DCPS::String type_name;
+  OpenDDS::XTypes::DynamicData temp_dd;
+  OpenDDS::XTypes::DynamicType_rch temp_dt;
+  switch (dt->get_kind()) {
+  case OpenDDS::XTypes::TK_INT8: {
+    ACE_CDR::Int8 my_int8;
+    if (dd.get_int8_value(my_int8, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_int8_value\n"), false);
```

Comment:
Please merge in `xtypes_dynamic` to get `log_level` and bring the logging in line with the new logging guidelines https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html#usage-in-opendds.